src/plugin/pluginControllers.ts

```diff
@@ -13,6 +24,97 @@ export const controllers: Record<PluginActionTypes, Resolver> = {
     });
   },
 
+  'sync-theme': () => {
+    const themePage = figma.root.children.find((node) => node.name === 'Theme');
+    const radiiFrame = themePage.findOne(
+      (node) => node.name === SliceFrameNames.Radius && node.type === 'FRAME'
+    ) as FrameNode;
+    const borderWidthsFrame = themePage.findOne(
+      (node) => node.name === SliceFrameNames.BorderWidth && node.type === 'FRAME'
+    ) as FrameNode;
+    const boxComponentId = themePage.getSharedPluginData(PLUGIN_DATA_NAMESPACE, PluginDataKeys.boxRefId);
+    const boxComponent = figma.getNodeById(boxComponentId);
+
+    if (!boxComponent || boxComponent.type !== 'COMPONENT_SET') {
+      figma.notify('Cant find the BOX primitive component', { error: true });
+      return;
+    }
+
+    const { existentRadiusSlices, newRadiusSlices } = syncRadiiVariants(radiiFrame);
+    const { existentBorderWidthSlices, newBorderWidthSlices } = syncBorderWidthVariants(borderWidthsFrame);
+
+    const newRadiiCombinations = getVariantCombinations([
+      { sliceName: Slices.Radius, styleKey: 'cornerRadius', variants: newRadiusSlices },
+      {
+        sliceName: Slices.BorderWidth,
+        styleKey: 'strokeWeight',
+        variants: existentBorderWidthSlices,
+      },
+    ]);
+
+    const newBorderWithCombinations = getVariantCombinations([
+      { sliceName: Slices.Radius, styleKey: 'cornerRadius', variants: { ...existentRadiusSlices, ...newRadiusSlices } },
+      {
+        sliceName: Slices.BorderWidth,
+        styleKey: 'strokeWeight',
+        variants: newBorderWidthSlices,
+      },
+    ]);
+
+    const newBoxVariants = createInstances([...newRadiiCombinations, ...newBorderWithCombinations]);
+
+    // add the new variants to the Box component
+    newBoxVariants.instances.map((newBoxVariant) => boxComponent.appendChild(newBoxVariant));
+
+    setRefs({ refIds: newBoxVariants.refIds?.[Slices.Radius], slices: radiiFrame.children });
+    setRefs({ refIds: newBoxVariants.refIds?.[Slices.BorderWidth], slices: borderWidthsFrame.children });
+
+    figma.notify('Components updated!');
+  },
+
+  'generate-theme-page': () => {
+    const page = figma.createPage();
+    page.name = 'Theme';
```

Comment:
Yes, if the user already have a page named "Theme" the plugin will create it anyway. But indeed that's bad since this name should be unique. I think I'll add this name to a constant, and change to "#Morfeo theme". It's also nice to add a check to prevent the plugin creating a duplication of this page

Reply:
So, we can name this page "Theme" if there are no other pages with this name. Otherwise "#Morfeo theme"

Reply:
Since we need to use the name as reference when sync slices&components, we should be sure what is the name, so:

- the name will be ever the same
- the user will be notified if a page with such name already exist (and the plugin will not create a new one)

Just committed those changes, check it out

Reply:
Well done ❤️